lib/matplotlib/backends/backend_macosx.py

```diff
@@ -66,7 +66,10 @@ class FigureManagerMac(_macosx.FigureManager, FigureManagerBase):
     Wrap everything up into a window for the pylab interface
     """
     def __init__(self, canvas, num):
-        _macosx.FigureManager.__init__(self, canvas)
+        icon_path = None
+        if 'com.adobe.pdf' in _macosx.FigureManager._get_image_types():
+            icon_path = str(cbook._get_data_path('images/matplotlib.pdf'))
+        _macosx.FigureManager.__init__(self, canvas, icon_path)
```

Comment:
My personal preference would be to keep it as you had it before with an explicit `_set_icon()` function rather than changing the init signature. But, I'm not sure I have a huge preference, others may have stronger feelings around this.

Reply:
I don't have strong feelings, I'll do whatever the consensus is. (Or just use your suggestion if I don't hear anything within a day or two.)